src/lib/pipeline/FetchUnit.cc

```diff
@@ -38,9 +38,18 @@ void FetchUnit::tick() {
     auto outputSlots = output_.getTailSlots();
     for (size_t slot = 0; slot < output_.getWidth(); slot++) {
       auto& macroOp = outputSlots[slot];
-      auto bytesRead = isa_.predecode(&(loopBuffer_.front().encoding),
-                                      loopBuffer_.front().instructionSize,
-                                      loopBuffer_.front().address, macroOp);
+      auto bytesRead = isa_.predecode(
+          reinterpret_cast<const uint8_t*>(&(loopBuffer_.front().encoding)),
+          loopBuffer_.front().instructionSize, loopBuffer_.front().address,
+          macroOp);
+
+      if (bytesRead == 0) {
+        std::cout << "[SimEng:FetchUnit] Predecode returned 0 bytes while loop "
+                     "buffer supplying"
+                  << std::endl;
+        exit(1);
+      }
+
       assert(bytesRead != 0 && "predecode failure for loop buffer entry");
```

Comment:
Is this needed anymore? If it is 0, the program will have exited above already.

Reply:
Good spot